---
title: Overview
sidebar: false
---

The single-cell workflow is comprised of multiple (standalone) subworkflows.

:::{.column-screen-inset-shaded}

```{mermaid}
%%| label: fig-architecture
%%| fig-cap: Overview single cell processing steps in OpenPipeline. Rectangles are data objects, parallelograms are Viash modules or subworkflows.

flowchart LR
  Raw[Raw\ndata]
  Dataset[Dataset]
  Ingestion[/Ingestion/]
  Preproc1[/Unimodal\nsingle-sample\nprocessing/]
  Preproc2[/Unimodal\nmulti-sample\nprocessing/]
  Concat[/Concatenating\nSamples/]
  Merge[/Merging\nmodes/]
  Integration[/Integration/]
  Interpretation[/Interpretation/]
  Conversion[/Conversion/]

  Raw --- Ingestion --> Split[/Split/] --> Preproc1 --> Concat --> Preproc2 --> Merge --> Integration --> Interpretation --> Conversion --> Dataset

  Ingestion -.-|"- Demux.\n- Mapping"| Ingestion
  Preproc1 -.-|"- Count QC\n- Doublet calling\n- Ambient RNA\n- Gatekeeper"| Preproc1
  Preproc2 -.-|"- Normalisation\n- Feature annotation\n- Feature selection\n- Batch correction?\n- Dim. red.?"| Preproc2
  Integration -.-|"- Dim. red.\n- Data integration"| Integration
  Interpretation -.-|"- Clustering\n- Cell typing\n- GRN\n- ..."| Interpretation

  linkStyle 10,11,12,13,14 stroke:#fff,stroke-width:0px,text-align:left;

  %% click Ingestion "./#ingestion"
  %% style Ingestion color:#2873c7,text-decoration:underline;
  %% click Preproc1 "./#single-sample-preproc"
  %% style Preproc1 color:#2873c7,text-decoration:underline;
  %% click Preproc2 "./#multi-sample-preproc"
  %% style Preproc2 color:#2873c7,text-decoration:underline;
  %% click Integration "./#integration"
  %% style Integration color:#2873c7,text-decoration:underline;
```

:::





## Subworkflows

### Ingestion

Purpose: Convert raw sequencing data or count tables into MuData data for further processing. 

:::{.column-screen-inset-shaded}

```{mermaid}
%%| label: fig-ingestion
%%| fig-cap: "Ingestion pipeline.  \n`*`: Possible entry points.  \n`†`: Output file(s)"
flowchart LR

  BCL["BCL<sup>*</sup>"]
  Fastq["Fastq<sup>*</sup>"]
  Ref["Reference"]
  RawDir["Raw out<sup>*</sup>"]
  RawCounts["Raw counts<sup>†</sup>"]

  Demux[/"Demux"/]
  Mapping[/"Mapping"/]
  Convert[/"Convert"/]

  BCL --- Demux --> Fastq
  Fastq & Ref --- Mapping --> RawDir --- Convert --> RawCounts


  BCL -.-|".tar.gz/Directory"| BCL
  Fastq -.-|".fastq.gz/.fastq"| Fastq
  Ref -.-|".tar.gz?"| Ref
  RawDir -.-|"Directory"| RawDir
  RawCounts -.-|".h5mu"| RawCounts

  linkStyle 7,8,9,10,11 stroke:#fff,stroke-width:0px,text-align:left;
```

:::


### Single-sample unimodal processing {#single-sample-preproc}

Purpose: Per modality fitering pipelines are available to select true from false cells. 

:::{.column-screen-inset-shaded}

```{mermaid}
%%| label: fig-sspreproc
%%| fig-cap: "Single-sample processing pipeline.  \n`*`: Possible entry points.  \n`†`: Output file(s)"
flowchart LR

  RawCounts["Raw counts<sup>*</sup>"]
  Processed["Processed\ncounts<sup>†</sup>"]

  Step1[/"Cell\nfiltering"/]
  Step2[/"Doublet\ncalling"/]
  Step3[/"Ambient RNA\ncorrection"/]
  
  RawCounts --> Step1 --> Step2 --> Step3 --> Processed
```

:::



### Multi-sample unimodal processing {#multi-sample-preproc}

Purpose: ... .

:::{.column-screen-inset-shaded}

```{mermaid}
%%| label: fig-mspreproc
%%| fig-cap: "Multi-sample processing pipeline.  \n`*`: Possible entry points.  \n`†`: Output file(s)"
flowchart LR

  Processed["Processed\ncounts<sup>*</sup>"]
  Normalised["Normalised\ncounts<sup>†</sup>"]

  Step1[/"Feature annotation"/]
  Step2[/"Batch correction?"/]
  Step3[/"Normalisation"/]
  Step4[/"Feature selection"/]

  Processed --> Step1 --> Step2 --> Step3 --> Step4 --> Normalised
```

:::



### Integration {#integration}

Purpose: Performs an integration pipeline for single cell data based on a single or multiple modalities. 

:::{.column-screen-inset-shaded}

```{mermaid}
%%| label: fig-integration
%%| fig-cap: "Integration pipeline.  \n`*`: Possible entry points.  \n`†`: Output file(s)"
flowchart LR

  Normalised["Normalised\ncounts<sup>*</sup>"]
  Integrated["Integrated\ndata<sup>†</sup>"]

  Step1[/"Data integration"/]
  Step2[/"Dimensionality\nreduction"/]
  
  Normalised --> Step1 --> Step2 --> Integrated
```

:::


### Interpretation {#interpretation}

Purpose: Take different dataset annotations and combine them together into a single enriched dataset. The idea is to have a diff_muon object, i.e. a muon object containing the changes of the original object where data from the diff_muon will be pushed to the original muon object. 


:::{.callout-note}
This is what we did a time ago and it has the drawback that it could make everything very slow. We might need to be able to aggregate diffs before adding them to the final object.
:::
