---
title: Components
---

## Workflows

:::{.column-body-outset}
|Namespace  |Name               |Description                                                                         |
|:----------|:------------------|:-----------------------------------------------------------------------------------|
|Ingestion  |[BD Rhapsody](workflow_ingestion_bd_rhapsody.qmd)|A BD Rhapsody mapping and conversion pipeline                                       |
|Ingestion  |[Cellranger mapping](workflow_ingestion_cellranger_mapping.qmd)|Cell Ranger mapping pipeline                                                        |
|Ingestion  |[Demux](workflow_ingestion_demux.qmd)|Convert bcl files to fastq files using bcl2fastq, bcl-convert or cellranger mkfastq |
|Multiomics |[Full pipeline](workflow_multiomics_full_pipeline.qmd)|A pipeline to analyse multiple multiomics samples                                   |
|Multiomics |[Integration](workflow_multiomics_integration.qmd)|Processing multimodal multi-sample RNA transcriptomics data                         |
|Multiomics |[Rna multisample](workflow_multiomics_rna_multisample.qmd)|Processing unimodal multi-sample RNA transcriptomics data                           |
|Multiomics |[Rna singlesample](workflow_multiomics_rna_singlesample.qmd)|Processing unimodal single-sample RNA transcriptomics data                          |
:::

## Components

:::{.column-body-outset}
|Namespace     |Name                                  |Description                                                                                                                          |
|:-------------|:-------------------------------------|:------------------------------------------------------------------------------------------------------------------------------------|
|Cluster       |[Leiden](module_cluster_leiden.qmd)   |Cluster cells using the Leiden algorithm [Traag18] implemented in the Scanpy framework [Wolf18]                                      |
|Convert       |[From 10xh5 to h5mu](module_convert_from_10xh5_to_h5mu.qmd)|Converts a 10x h5 into an h5mu file                                                                                                  |
|Convert       |[From 10xmtx to h5mu](module_convert_from_10xmtx_to_h5mu.qmd)|Converts a 10x mtx into an h5mu file                                                                                                 |
|Convert       |[From bd to 10x molecular barcode tags](module_convert_from_bd_to_10x_molecular_barcode_tags.qmd)|Convert the molecular barcode sequence SAM tag from BD format (MA) to 10X format (UB)                                                |
|Convert       |[From bdrhap to h5mu](module_convert_from_bdrhap_to_h5mu.qmd)|Convert the output of a BD Rhapsody WTA pipeline to a MuData h5 file                                                                 |
|Convert       |[From h5ad to h5mu](module_convert_from_h5ad_to_h5mu.qmd)|Converts a single layer h5ad file into a single muon object                                                                          |
|Convert       |[From h5mu to seurat](module_convert_from_h5mu_to_seurat.qmd)|Converts an h5mu file into a Seurat file                                                                                             |
|Convert       |[From velocyto to h5mu](module_convert_from_velocyto_to_h5mu.qmd)|Convert a velocyto                                                                                                                   |
|Demux         |[Bcl convert](module_demux_bcl_convert.qmd)|Convert bcl files to fastq files using bcl-convert                                                                                   |
|Demux         |[Bcl2fastq](module_demux_bcl2fastq.qmd)|Convert bcl files to fastq files using bcl2fastq                                                                                     |
|Demux         |[Cellranger mkfastq](module_demux_cellranger_mkfastq.qmd)|Demultiplex raw sequencing data                                                                                                      |
|Dimred        |[Pca](module_dimred_pca.qmd)          |Computes PCA coordinates, loadings and variance decomposition                                                                        |
|Dimred        |[Umap](module_dimred_umap.qmd)        |UMAP (Uniform Manifold Approximation and Projection) is a manifold learning technique suitable for visualizing high-dimensional data |
|Download      |[Download file](module_download_download_file.qmd)|Download a file                                                                                                                      |
|Download      |[Sync test resources](module_download_sync_test_resources.qmd)|Synchronise the test resources from s3://openpipelines-data to resources_test                                                        |
|Files         |[Make params](module_files_make_params.qmd)|Looks for files in a directory and turn it in a params file                                                                          |
|Filter        |[Do filter](module_filter_do_filter.qmd)|Remove observations and variables based on specified                                                                                 |
|Filter        |[Filter with counts](module_filter_filter_with_counts.qmd)|Filter scRNA-seq data based on the primary QC metrics                                                                                |
|Filter        |[Filter with hvg](module_filter_filter_with_hvg.qmd)|Annotate highly variable genes [Satija15] [Zheng17] [Stuart19]                                                                       |
|Filter        |[Filter with scrublet](module_filter_filter_with_scrublet.qmd)|Doublet detection using the Scrublet method (Wolock, Lopez and Klein, 2019)                                                          |
|Integrate     |[Concat](module_integrate_concat.qmd) |Concatenates several uni-modal samples in                                                                                            |
|Integrate     |[Harmony](module_integrate_harmony.qmd)|Performs Harmony integration based as described in https://github                                                                    |
|Integrate     |[Harmonypy](module_integrate_harmonypy.qmd)|Performs Harmony integration based as described in https://github                                                                    |
|Integrate     |[Merge](module_integrate_merge.qmd)   |Combine one or more single-modality                                                                                                  |
|Integrate     |[Scanorama](module_integrate_scanorama.qmd)|Use Scanorama to integrate different experiments                                                                                     |
|Mapping       |[BD Rhapsody](module_mapping_bd_rhapsody.qmd)|A wrapper for the BD Rhapsody Analysis CWL v1.10.1 pipeline                                                                          |
|Mapping       |[Cellranger count split](module_mapping_cellranger_count_split.qmd)|Split 10x Cell Ranger output directory into separate output fields                                                                   |
|Mapping       |[Cellranger count](module_mapping_cellranger_count.qmd)|Align fastq files using Cell Ranger count                                                                                            |
|Neighbors     |[Bbknn](module_neighbors_bbknn.qmd)   |BBKNN network generation                                                                                                             |
|Neighbors     |[Find neighbors](module_neighbors_find_neighbors.qmd)|Compute a neighborhood graph of observations [McInnes18]                                                                             |
|Process 10xh5 |[Filter 10xh5](module_process_10xh5_filter_10xh5.qmd)|Filter a 10x h5 dataset                                                                                                              |
|Qc            |[Fastqc](module_qc_fastqc.qmd)        |Fastqc component, please see https://www                                                                                             |
|Qc            |[Multiqc](module_qc_multiqc.qmd)      |Component for multiqc (https://multiqc                                                                                               |
|Reference     |[Build bdrhap reference](module_reference_build_bdrhap_reference.qmd)|Compile a reference into a STAR index compatible with the BD Rhapsody pipeline                                                       |
|Reference     |[Build cellranger reference](module_reference_build_cellranger_reference.qmd)|Build a Cell Ranger-compatible reference folder from user-supplied genome FASTA and gene GTF files                                   |
|Reference     |[Make reference](module_reference_make_reference.qmd)|Make a reference build                                                                                                               |
|Report        |[Mermaid](module_report_mermaid.qmd)  |Generates a network from mermaid code                                                                                                |
|Split         |[Split modalities](module_split_split_modalities.qmd)|Split the modalities from a single                                                                                                   |
|Transfer      |[Publish](module_transfer_publish.qmd)|Publish an artifact and optionally rename with parameters                                                                            |
|Transform     |[Delete layer](module_transform_delete_layer.qmd)|Delete an anndata layer from one or more modalities                                                                                  |
|Transform     |[Log1p](module_transform_log1p.qmd)   |Logarithmize the data matrix                                                                                                         |
|Transform     |[Normalize total](module_transform_normalize_total.qmd)|Normalize counts per cell                                                                                                            |
|Transform     |[Regress out](module_transform_regress_out.qmd)|Regress out (mostly) unwanted sources of variation                                                                                   |
|Transform     |[Scale](module_transform_scale.qmd)   |Scale data to unit variance and zero mean                                                                                            |
|Velocity      |[Scvelo](module_velocity_scvelo.qmd)  |                                                                                                                                     |
|Velocity      |[Velocyto](module_velocity_velocyto.qmd)|Runs the velocity analysis on a BAM file, outputting a loom file                                                                     |
:::
