---
title: "Cellbender remove background"
toc: false
---

::: {.column-margin}

### Info

ID: `cellbender_remove_background`  
Namespace: `correction`

### Links
[Source](https://github.com/openpipelines-bio/openpipeline/blob/main/src/correction/cellbender_remove_background){.btn-action .btn-sm .btn-info .btn role="button"}

:::

Eliminating technical artifacts from high-throughput single-cell RNA sequencing data.

This module removes counts due to ambient RNA molecules and random barcode swapping from (raw) UMI-based scRNA-seq count matrices. 
At the moment, only the count matrices produced by the CellRanger count pipeline is supported. Support for additional tools and protocols 
will be added in the future. A quick start tutorial can be found here.

Fleming et al. 2022, bioRxiv.


## Example commands

You can run the pipeline using `nextflow run`.

### View help

You can use `--help` as a parameter to get an overview of the possible parameters.

```bash
nextflow run openpipelines-bio/openpipeline \
  -r main_build -latest \
  -main-script target/nextflow/correction/cellbender_remove_background/main.nf \
  --help
```

### Run command

<details>
<summary>Example of `params.yaml`</summary>
```yaml
# Inputs
input: # please fill in - example: "input.h5"
modality: "rna"

# Outputs
# output: "$id.$key.output.h5"
layer_output: "corrected"
obs_latent_rt_efficiency: "latent_rt_efficiency"
obs_latent_cell_probability: "latent_cell_probability"
obs_latent_scale: "latent_scale"
var_ambient_expression: "ambient_expression"

# Arguments
# model: "foo"
# total_droplets_included: 50000
# epochs: 150
# fpr: 0.01
exclude_antibody_capture: false
# learning_rate: 1.0E-4
cuda: false

# Nextflow input-output arguments
publish_dir: # please fill in - example: "output/"
# param_list: "my_params.yaml"
```
</details>

```bash
nextflow run openpipelines-bio/openpipeline \
  -r main_build -latest \
  -profile docker \
  -main-script target/nextflow/correction/cellbender_remove_background/main.nf \
  -params-file params.yaml
```

:::{.callout-note}
Replace `-profile docker` with `-profile podman` or `-profile singularity` depending on the desired backend.
:::

## Argument groups

### Inputs

|Name         |Description                    |Attributes                              |
|:------------|:------------------------------|:---------------------------------------|
|`--input`    |Input file.                    |`file`, required, example: `"input.h5"` |
|`--modality` |List of modalities to process. |`string`, default: `"rna"`              |


### Outputs

|Name             |Description                                                                                                           |Attributes                               |
|:----------------|:---------------------------------------------------------------------------------------------------------------------|:----------------------------------------|
|`--output`       |Full count matrix as an h5mu file, with background RNA removed. This file contains all the original droplet barcodes. |`file`, required, example: `"output.h5"` |
|`--layer_output` |Output layer                                                                                                          |`string`, default: `"corrected"`         |


### Arguments

|Name                         |Description                                                                                                                                                                                                                                                                                                                                                               |Attributes                  |
|:----------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:---------------------------|
|`--model`                    |Which model is being used for count data. 'simple' does not model either ambient RNA or random barcode swapping (for debugging purposes -- not recommended). 'ambient' assumes background RNA is incorporated into droplets. 'swapping' assumes background RNA comes from random barcode swapping. 'full' uses a combined ambient and swapping model. Defaults to 'full'. |`string`                    |
|`--total_droplets_included`  |The number of droplets from the rank-ordered UMI plot that will be analyzed. The largest 'total_droplets' droplets will have their cell probabilities inferred as an output.                                                                                                                                                                                              |`integer`, example: `50000` |
|`--epochs`                   |Number of epochs to train                                                                                                                                                                                                                                                                                                                                                 |`integer`, example: `150`   |
|`--fpr`                      |Target false positive rate in (0, 1). A false positive is a true signal count that is erroneously removed. More background removal is accompanied by more signal removal at high values of FPR. You can specify multiple values, which will create multiple output files.                                                                                                 |`double`, example: `0.01`   |
|`--exclude_antibody_capture` |Including the flag --exclude-antibody-capture will cause remove-background to operate on gene counts only, ignoring other features.                                                                                                                                                                                                                                       |`boolean_true`              |
|`--learning_rate`            |Training detail: lower learning rate for inference. A OneCycle learning rate schedule is used, where the upper learning rate is ten times this value. (For this value, probably do not exceed 1e-3).                                                                                                                                                                      |`double`, example: `1e-04`  |
|`--cuda`                     |Including the flag --cuda will run the inference on a GPU.                                                                                                                                                                                                                                                                                                                |`boolean_true`              |

## Authors

