
---
title: "bd_rhapsody"
---

## Pipeline arguments

nextflow run \
https://github.com/openpipelines-bio/openpipeline.git -r main_build \
-main-script workflows/ingestion/bd_rhapsody/main.nf \
--help

## Table of arguments

| name                       | type   | description                                                                                                                                                                                                           |
|:---------------------------|:-------|:----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| --mode                     | string | Whether to run a whole transcriptome analysis (WTA) or a targeted analysis.                                                                                                                                           |
| --id                       | string | ID of the sample.                                                                                                                                                                                                     |
| --input                    | file   | Path to your read files in the FASTQ.GZ format. You may specify as many R1/R2 read pairs as you want.                                                                                                                 |
| --reference                | file   | Refence to map to. For `--mode wta`, this is the path to STAR index as a tar.gz file. For `--mode targeted`, this is the path to mRNA reference file for pre-designed, supplemental, or custom panel, in FASTA format |
| --transcriptome_annotation | file   | Path to GTF annotation file (only for `--mode wta`).                                                                                                                                                                  |
| --abseq_reference          | file   | Path to the AbSeq reference file in FASTA format. Only needed if BD AbSeq Ab-Oligos are used.                                                                                                                         |
| --supplemental_reference   | file   | Path to the supplemental reference file in FASTA format. Only needed if there are additional transgene sequences used in the experiment (only for `--mode wta`).                                                      |
| --sample_prefix            | string | Specify a run name to use as the output file base name. Use only letters, numbers, or hyphens. Do not use special characters or spaces.                                                                               |
| name     | type   | description                                                |
|:---------|:-------|:-----------------------------------------------------------|
| --output | file   | Output folder. Output still needs to be processed further. |
| name                       | type         | description                                                                                                                                                                                                                                                                                                                                                     |
|:---------------------------|:-------------|:----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| --putative_cell_call       | string       | Specify the dataset to be used for putative cell calling. For putative cell calling using an AbSeq dataset, please provide an AbSeq_Reference fasta file above.                                                                                                                                                                                                 |
| --exact_cell_count         | integer      | Exact cell count - Set a specific number (>=1) of cells as putative, based on those with the highest error-corrected read count                                                                                                                                                                                                                                 |
| --disable_putative_calling | boolean_true | Disable Refined Putative Cell Calling - Determine putative cells using only the basic algorithm (minimum second derivative along the cumulative reads curve). The refined algorithm attempts to remove false positives and recover false negatives, but may not be ideal for certain complex mixtures of cell types. Does not apply if Exact Cell Count is set. |
| name             | type    | description                                                                                     |
|:-----------------|:--------|:------------------------------------------------------------------------------------------------|
| --subsample      | double  | A number >1 or fraction (0 < n < 1) to indicate the number or percentage of reads to subsample. |
| --subsample_seed | integer | A seed for replicating a previous subsampled run.                                               |
| name                  | type   | description                                                                                                                    |
|:----------------------|:-------|:-------------------------------------------------------------------------------------------------------------------------------|
| --sample_tags_version | string | Specify if multiplexed run.                                                                                                    |
| --tag_names           | string | Tag_Names (optional) - Specify the tag number followed by '-' and the desired sample name to appear in Sample_Tag_Metrics.csv. |
|                       |        | Do not use the special characters: &, (), [], {},  <>, ?, |                                                                    |
| name          | type   | description         |
|:--------------|:-------|:--------------------|
| --vdj_version | string | Specify if VDJ run. |
| name         | type         | description                                                                                                        |
|:-------------|:-------------|:-------------------------------------------------------------------------------------------------------------------|
| --parallel   | boolean      | Run jobs in parallel.                                                                                              |
| --timestamps | boolean_true | Add timestamps to the errors, warnings, and notifications.                                                         |
| --dryrun     | boolean_true | If true, the output directory will only contain the CWL input files, but the pipeline itself will not be executed. |